containers/orchestration/app/main.py

```
@@ -0,0 +1,341 @@
+        data = {"data": r["response"]["FhirResource"]}
+    elif "geocode" in step["endpoint"]:
+        data = {
+            "bundle": r["bundle"],
+            "geocode_method": processing_config["configurations"][
+                "standardization_and_geocoding"
+            ]["geocode_method"],
+            "smarty_auth_id": os.environ.get("SMARTY_AUTH_ID"),
+            "smarty_auth_token": os.environ.get("SMARTY_AUTH_TOKEN"),
+            "license_type": os.environ.get("LICENSE_TYPE"),
+        }
+    else:
+        data = {"data": r["bundle"]}
+
+    ingestion_response = requests.post(ingestion_url + step["endpoint"], json=data)
+    return ingestion_response
+
+
+def call_message_parser(input, response, step) -> dict:
+    r = response.json()
+    data = {
+        "message_format": processing_config["configurations"]["message_parser"][
+            "message_format"
+        ],
+        "parsing_schema_name": processing_config["configurations"]["message_parser"][
+            "parsing_schema_name"
+        ],
+        "message": r["bundle"],
+    }
+
+    message_parser_response = requests.post(
+        message_parser_url + step["endpoint"], json=data
+    )
+    return message_parser_response
+
+
+@app.post("/process", status_code=200, responses=process_message_response_examples)
+async def process_message_endpoint(
+    input: Annotated[
+        ProcessMessageRequest, Body(examples=process_message_request_examples)
+    ]
+) -> ProcessMessageResponse:
+    """
+    Process message through a series of microservices
+    """
+    # Change below to grab from uploaded configs once we've got them
+    processing_config = load_processing_config("sample-orchestration-config.json")
+    input = dict(input)
+    response = input
+    responses = {}
+
+    for step in processing_config["steps"]:
+        service = step["service"]
+        endpoint = step["endpoint"]
+        f = f"call_{service}"
+        if f in globals() and callable(globals()[f]):
```

very neat! one minor suggestion would be to pull this "find a function" logic out to a helper method - it takes a few reads to understand that it's going off to search for `call_$service_name`.

---

Also, what's the expected behavior if the function isn't found? I would think we'd need to throw some kind of error back to the user - "requested function not found" or similar.